system/lib/pthread/pthread_create.c

```diff
@@ -68,7 +68,35 @@ static long dummy_getpid() {
 }
 weak_alias(dummy_getpid, __syscall_getpid);
 
-int __pthread_create(pthread_t *restrict res, const pthread_attr_t *restrict attrp, void *(*entry)(void *), void *restrict arg) {
+static int tl_lock_count;
```

Comment:
Is all this logic copied from a musl file? If so which?

Reply:
Yes, there is already comment at the top saying ` // See musl's pthread_create.c`.  Hopefully we can do a better job of code sharing as a followup.